rust/migration/src/m20231117_162713_add_created_at.rs

```diff
@@ -0,0 +1,112 @@
+use sea_orm_migration::prelude::*;
+
+#[derive(DeriveMigrationName)]
+pub struct Migration;
+
+#[macro_export]
+macro_rules! static_for {
+    ({ $( $x:expr ),* } as $var:pat_param in $body:block) => {
+        {
+            $(
+                {
+                    let $var = $x;
+                    $body;
+                }
+            )*
+        }
+    };
+}
+
+#[async_trait::async_trait]
+impl MigrationTrait for Migration {
+    async fn up(&self, manager: &SchemaManager) -> Result<(), DbErr> {
+        static_for!({
+          (Job::Table, Job::CreatedAt),
+          (OutputDir::Table, OutputDir::CreatedAt),
+          (OutputSymlink::Table, OutputSymlink::CreatedAt),
+          (OutputFile::Table, OutputFile::CreatedAt),
+          (VisibleFile::Table, VisibleFile::CreatedAt),
+          (ApiKey::Table, ApiKey::CreatedAt),
+          (JobUses::Table, JobUses::CreatedAt)
+        } as (t, c) in {
+            manager
+                .alter_table(
+                    Table::alter()
+                        .table(t)
+                        .add_column(
+                            ColumnDef::new(c)
+                                .timestamp()
+                                .not_null()
+                                .default(SimpleExpr::Keyword(Keyword::CurrentTimestamp))
```

Comment:
It took me seeing the NotSet to realize that how nice this was. I thought it was setting the infill for the migration but it makes the other insertions automatically set it to?

Something is going on with SQL there that I don't understand lol

Reply:
> but it makes the other insertions automatically set it to

Correct, that column is automatically filled by the db engine

Reply:
This is just me being curious about SQL: What happens if you do `2 * CurrentTimestamp`? Is the default allowed to be a *function* of the current time?! Or is there just a special keyword in SQL that I'm not familiar with and something will error out if you try to use a function in that context?

Reply:
its a function but yeah https://www.postgresqltutorial.com/postgresql-date-functions/postgresql-current_timestamp/

Reply:
ok this clears things up for me: https://www.postgresql.org/docs/current/ddl-default.html

> The default value can be an expression, which will be evaluated whenever the default value is inserted (not when the table is created). A common example is for a timestamp column to have a default of CURRENT_TIMESTAMP, so that it gets set to the time of row insertion. Another common example is generating a “serial number” for each row. In PostgreSQL this is typically done by something like:

Reply:
I didn't know this was possible, TIL!